pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -238,15 +242,22 @@ func (ev *Evaluator) findCandidates(ctx context.Context, allNodes []*framework.N
 		return nil, nil, err
 	}
 
-	offset, numCandidates := ev.GetOffsetAndNumCandidates(int32(len(potentialNodes)))
+	offset, candidatesNum := ev.GetOffsetAndNumCandidates(int32(len(potentialNodes)))
 	if loggerV := logger.V(5); logger.Enabled() {
 		var sample []string
-		for i := offset; i < offset+10 && i < int32(len(potentialNodes)); i++ {
-			sample = append(sample, potentialNodes[i].Node().Name)
+		if sampleLength >= len(potentialNodes) {
+			for _, node := range potentialNodes {
+				sample = append(sample, node.Node().Name)
+			}
+		} else {
+			for i := 0; i < sampleLength; i++ {
+				nodeName := potentialNodes[(int(offset)+i)%len(potentialNodes)].Node().Name
+				sample = append(sample, nodeName)
+			}
 		}
-		loggerV.Info("Selected candidates from a pool of nodes", "potentialNodesCount", len(potentialNodes), "offset", offset, "sampleLength", len(sample), "sample", sample, "candidates", numCandidates)
```

Comment:
can't we simply output all nodes, in the first place?

Reply:
In case too many candidates? which will lead to performance issue.

Reply:
Is the log that only contains the part of candidates worthwhile for debugging actually?

I suggest either increasing the log level and including all nodes, or just removeing `sample` from the log and adding a new log in `checkNode` in `DryRunPreemption` so that we can print all node names without any new for-loop for candidates.

Reply:
Not a good idea to print all the nodes here just because too many unschedulable nodes will still lead to performance issue, although we increase the log level. But log the message in `DryRunPreemption` sounds like a good suggestion.

Reply:
Done and squashed.